### PR TITLE
Migrations without model

### PR DESCRIPTION
**Pull request by alneberg**
- [x] Tests passing
- [x] Black formatting
- [x] Migrations for any changes to the database schema
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

I've moved to using alembic directly. Some issues that I've tried to fix:
 - A column can not be created as non-nullable directly, since existing data would cause it to fail
 - A column can not be changed to non-nullable without first handling the null values
 - Removed apscheduler from an existing downgrade version

Also something that needs to be checked extra carefully:
 - Changed from `sqlalchemy.boolean` to `mysql.TINYINT` in existing migration (This might cause issues with existing data?)

Manual tests performed:

```
git checkout dev -- migrations

docker-compose --profile cli down
docker image rm mariadb (just in case)
docker-compose up --build

docker exec -it dds_backend /bin/bash
apt-get install -y mariadb-client
mysqldump -h db -u root -pTEST_PASSWORD DeliverySystem > original_db.sql
exit

git reset HEAD migrations
git checkout -- migrations


flask db upgrade
[2022-03-10 08:16:57,094] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
root@0c2411fb47b0:/code# flask db downgrade
[2022-03-10 08:17:04,727] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade 666003748d14 -> a5a40d843415, change_active_nullable_and_default
root@0c2411fb47b0:/code# flask db downgrade
[2022-03-10 08:17:12,151] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade a5a40d843415 -> d117e6299dc9, changed_sensitive
root@0c2411fb47b0:/code# flask db downgrade
[2022-03-10 08:17:32,368] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade d117e6299dc9 -> e5a51e5ed9a1, remove_project-invites
root@0c2411fb47b0:/code# flask db upgrade
[2022-03-10 08:22:27,314] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade e5a51e5ed9a1 -> d117e6299dc9, remove_project-invites
INFO  [alembic.runtime.migration] Running upgrade d117e6299dc9 -> a5a40d843415, changed_sensitive
INFO  [alembic.runtime.migration] Running upgrade a5a40d843415 -> 666003748d14, change_active_nullable_and_default
```

```
mysqldump -h db -u root -pTEST_PASSWORD DeliverySystem > second_db.sql

diff original_db.sql second_db.sql
```
The last one needs manual inspection and is very messy due to the random bytes in the test data.

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1040"></a>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 - Add support for getting IPs from X-Forwarded-For ([#952](https://github.com/ScilifelabDataCentre/dds_web/pull/952))
 - Relax requirements for usernames (wider length range, `.` and `-`) ([#943](https://github.com/ScilifelabDataCentre/dds_web/pull/943))
 - Delay committing project to db until after the bucket has been created ([#967](https://github.com/ScilifelabDataCentre/dds_web/pull/967))
-- Fix logic for notification about sent email ([#963])(https://github.com/ScilifelabDataCentre/dds_web/pull/963))
+- Fix logic for notification about sent email ([#963])(<https://github.com/ScilifelabDataCentre/dds_web/pull/963>))
```
Comment (inaod568) [inline CHANGELOG.md:32]:
```suggestion
- Fix logic for notification about sent email ([#963])(https://github.com/ScilifelabDataCentre/dds_web/pull/963))
```

Comment (alneberg) [reply CHANGELOG.md]: Thanks!

**Comment by inaod568** [comment]
@alneberg @talavis So I tested now. I did all the steps you showed above, but after that I did this and got errors: 
```
(dds_venv) $ docker exec dds_backend flask db downgrade
[2022-03-11 10:46:04,979] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade 666003748d14 -> a5a40d843415, change_active_nullable_and_default

[manually set active for a user to null - successful, no issues]

(dds_venv) $ docker exec dds_backend flask db upgrade
[2022-03-11 10:48:41,154] __init__ [INFO] Logging initiated.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade a5a40d843415 -> 666003748d14, change_active_nullable_and_default
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.10/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.10/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.10/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.10/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1265, "Data truncated for column 'active' at row 5")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/flask", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 994, in main
    cli.main(args=sys.argv[1:])
  File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 600, in main
    return super().main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 444, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask_migrate/cli.py", line 149, in upgrade
    _upgrade(directory, revision, sql, tag, x_arg)
  File "/usr/local/lib/python3.10/site-packages/flask_migrate/__init__.py", line 98, in wrapped
    f(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask_migrate/__init__.py", line 185, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/usr/local/lib/python3.10/site-packages/alembic/command.py", line 320, in upgrade
    script.run_env()
  File "/usr/local/lib/python3.10/site-packages/alembic/script/base.py", line 563, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/usr/local/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 92, in load_python_file
    module = load_module_py(module_id, path)
  File "/usr/local/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 108, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/code/migrations/env.py", line 88, in <module>
    run_migrations_online()
  File "/code/migrations/env.py", line 82, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/usr/local/lib/python3.10/site-packages/alembic/runtime/environment.py", line 851, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/usr/local/lib/python3.10/site-packages/alembic/runtime/migration.py", line 620, in run_migrations
    step.migration_fn(**kw)
  File "/code/migrations/versions/666003748d14_change_active_nullable_and_default.py", line 24, in upgrade
    op.alter_column("users", "active", existing_type=mysql.TINYINT(display_width=1), nullable=False)
  File "<string>", line 8, in alter_column
  File "<string>", line 3, in alter_column
  File "/usr/local/lib/python3.10/site-packages/alembic/operations/ops.py", line 1880, in alter_column
    return operations.invoke(alt)
  File "/usr/local/lib/python3.10/site-packages/alembic/operations/base.py", line 392, in invoke
    return fn(self, operation)
  File "/usr/local/lib/python3.10/site-packages/alembic/operations/toimpl.py", line 50, in alter_column
    operations.impl.alter_column(
  File "/usr/local/lib/python3.10/site-packages/alembic/ddl/mysql.py", line 116, in alter_column
    self._exec(
  File "/usr/local/lib/python3.10/site-packages/alembic/ddl/impl.py", line 193, in _exec
    return conn.execute(construct, multiparams)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1289, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/sql/ddl.py", line 80, in _execute_on_connection
    return connection._execute_ddl(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1381, in _execute_ddl
    ret = self._execute_context(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1845, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2026, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.10/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.10/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.10/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.10/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.10/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
sqlalchemy.exc.DataError: (pymysql.err.DataError) (1265, "Data truncated for column 'active' at row 5")
[SQL: ALTER TABLE users MODIFY active TINYINT(1) NOT NULL]
(Background on this error at: https://sqlalche.me/e/14/9h9h)
```

File: migrations/versions/666003748d14_change_active_nullable_and_default.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 def upgrade():
     # ### commands auto generated by Alembic - please adjust! ###
     user_table = sa.sql.table("users", sa.sql.column("active", mysql.TINYINT(display_width=1)))
-    op.execute(user_table.update().where(user_table.c.active is None).values(active=False))
+    op.execute(user_table.update().where(user_table.c.active == None).values(active=False))
```
Comment (inaod568) [inline migrations/versions/666003748d14_change_active_nullable_and_default.py:23]: Huh, interesting! I'll try again now

Comment (inaod568) [reply migrations/versions/666003748d14_change_active_nullable_and_default.py:23]: Works now!

**Review by inaod568 — APPROVED** [review]
Looks good and works well, so if you @alneberg and @talavis is fine with this I'm ready to merge.

**Comment by inaod568** [comment]
Since this affects all previous migrations I want to wait until @talavis has also approved this, and he is currently vabbing.

**Review by talavis — APPROVED** [review]
Seems to work as intended. Tested downgrade/upgrade with the standard test data.

**Review by inaod568 — APPROVED** [review]

**Pull request merged**